app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,13 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+
+    if @user.persisted?
+      flash[:notice] = I18n.t "devise.omniauth_callbacks.success", kind: "Google"
+      sign_in_and_redirect @user, event: :authentication
+    else
+      session["devise.google_data"] = request.env["omniauth.auth"]
```

Comment:
I think, i can redirect to new_page, if user have errors, and can't be saved. But here i doesn't render user errors ((. redirect_to_registration_path with alert !